src/rng.rs

```diff
@@ -0,0 +1,115 @@
+use core::cmp;
+use core::mem;
+
+use crate::hal::blocking::rng;
+use crate::stm32::RNG;
+use crate::stm32;
+
+#[derive(Debug)]
+pub enum ErrorKind {
+    ClockError,
+    SeedError,
+}
+
+
+pub trait RngExt {
+    fn constrain(self) -> Rng;
+}
+
+impl RngExt for RNG {
+
+    /// Enable RNG_CLK and the RNG peripheral.
+    /// Note that clocks must already be configured
+    /// such that RNG_CLK is not less than 1/16 HCLK,
+    /// otherwise all reads of the RNG will return
+    /// a ClockError (CECS error)
+    fn constrain(self) -> Rng {
```

Comment:
We should probably take the clock configuration here and make sure the prerequisites for the clock speeds are met.